tests/datasource/data_connector/test_configured_asset_azure_data_connector.py

```diff
@@ -0,0 +1,1576 @@
+)
+@mock.patch(
+    "great_expectations.core.usage_statistics.usage_statistics.UsageStatisticsHandler.emit"
+)
+def test_return_all_batch_definitions_too_many_sorters(
+    mock_azure_conn, mock_list_keys, mock_emit, empty_data_context_stats_enabled
+):
+    my_data_connector_yaml = yaml.load(
+        f"""
+       class_name: ConfiguredAssetAzureDataConnector
+       datasource_name: test_environment
+       execution_engine:
+           class_name: PandasExecutionEngine
+       container: my_container
+       name_starts_with: ""
+       assets:
+           TestFiles:
+       default_regex:
+           pattern: (.+)_.+_.+\\.csv
+           group_names:
+               - name
+       sorters:
+           - orderby: asc
+             class_name: LexicographicSorter
+             name: name
+           - datetime_format: "%Y%m%d"
+             orderby: desc
+             class_name: DateTimeSorter
+             name: timestamp
+           - orderby: desc
+             class_name: NumericSorter
+             name: price
+
+       azure_options:
+           account_url: my_account_url.blob.core.windows.net
+           credential: my_credential
+   """,
+    )
+
+    mock_list_keys.return_value = [
+        "alex_20200809_1000.csv",
+        "eugene_20200809_1500.csv",
+        "james_20200811_1009.csv",
+        "abe_20200809_1040.csv",
+        "will_20200809_1002.csv",
+        "james_20200713_1567.csv",
+        "eugene_20201129_1900.csv",
+        "will_20200810_1001.csv",
+        "james_20200810_1003.csv",
+        "alex_20200819_1300.csv",
+    ]
+
+    with pytest.raises(ge_exceptions.DataConnectorError):
+        instantiate_class_from_config(
+            config=my_data_connector_yaml,
+            runtime_environment={
+                "name": "general_azure_data_connector",
+                "datasource_name": "test_environment",
+            },
+            config_defaults={
+                "module_name": "great_expectations.datasource.data_connector"
+            },
+        )
+
+
+@mock.patch(
+    "great_expectations.datasource.data_connector.configured_asset_azure_data_connector.BlobServiceClient"
+)
+@mock.patch(
+    "great_expectations.datasource.data_connector.configured_asset_azure_data_connector.list_azure_keys",
+)
+@mock.patch(
+    "great_expectations.core.usage_statistics.usage_statistics.UsageStatisticsHandler.emit"
+)
+def test_example_with_explicit_data_asset_names(
+    mock_azure_conn, mock_list_keys, mock_emit, empty_data_context_stats_enabled
+):
+    yaml_string = f"""
+class_name: ConfiguredAssetAzureDataConnector
+datasource_name: FAKE_DATASOURCE_NAME
+container: my_container
+name_starts_with: my_base_directory/
+default_regex:
+   pattern: ^(.+)-(\\d{{4}})(\\d{{2}})\\.(csv|txt)$
+   group_names:
+       - data_asset_name
+       - year_dir
+       - month_dir
+assets:
+   alpha:
+       prefix: my_base_directory/alpha/files/go/here/
```

Comment:
Do we need to add the Azure-specific keys to the `Asset` schema in `base.py` (e.g., `container`)?

Reply:
ah yes they've already been added (but not reflected here). Great catch 🙇🏽